packages/odyssey-react-mui/src/labs/SideNav/SideNav.tsx

```diff
@@ -333,62 +381,43 @@ const SideNav = ({
       sideNavItems.map((item) => ({
         ...item,
         children: item.children?.map((childProps) => (
-          <SideNavItemContent
+          <SideNavItemContentContext.Provider
+            value={{ isCompact: isCompact }}
             key={childProps.id}
-            scrollRef={getRefIfThisIsFirstNodeWithIsSelected(childProps.id)}
-            {...childProps}
-          />
+          >
+            <SideNavItemContent
+              key={childProps.id}
+              scrollRef={getRefIfThisIsFirstNodeWithIsSelected(childProps.id)}
+              {...childProps}
+            />
```

Comment:
I prefer to have `{...childProps}` first because it's more clear what's going to happen. The way this is written, I can never tell if `scrollRef` is gonna get overridden.

```suggestion
            <SideNavItemContent
              {...childProps}
              key={childProps.id}
              scrollRef={getRefIfThisIsFirstNodeWithIsSelected(childProps.id)}
            />
```